front-end-tools/src/components/InitComponent.tsx

```diff
@@ -265,107 +305,125 @@ export default function InitComponent(props: ConnectionProps) {
+                                </div>
+                            </Alert>
+                            <Alert variant="info" style={{ textAlign: 'left' }}>
+                                <ul>
+                                    <li>
+                                        Select <b>{DO_NOT_DERIVE.label}</b> in the above drop down, if you want to
+                                        manually fill in the <code>smart contract name</code>, or an{' '}
+                                        <code>input parameter schema</code>.
+                                    </li>
+                                    <li>
+                                        Select <b>{DERIVE_FROM_CHAIN.label}</b> again in the drop down, if you want to
+                                        load a new module from the chain because this box will not automatically update
+                                        when you change the moduel reference in field below.
+                                    </li>
+                                </ul>
+                            </Alert>
+                            <br />
+                        </>
+                    )}
+                </Form.Group>
 
                 {moduleReferenceError && <Alert variant="danger"> Error: {moduleReferenceError}. </Alert>}
-
+                <br />
                 <Row>
                     <Form.Group className="col-md-4 mb-3">
                         <Form.Label> Module Reference</Form.Label>
                         <Form.Control
-                            defaultValue="91225f9538ac2903466cc4ab07b6eb607a2cd349549f357dfdf4e6042dde0693"
-                            disabled={!!useModuleReferenceFromStep1}
-                            {...form.register('moduleReference', { required: true })}
-                            onChange={(e) => {
-                                const register = form.register('moduleReference', {
-                                    required: true,
-                                });
-
-                                register.onChange(e);
-
-                                setModuleReferenceLengthError(undefined);
-
-                                const moduleRef = form.getValues('moduleReference');
-
-                                if (moduleRef !== undefined && !REG_MODULE_REF.test(moduleRef)) {
-                                    setModuleReferenceLengthError(
-                                        'Module reference has to be a valid hex string `[0-9A-Fa-f]` of length 64'
-                                    );
-                                }
-                            }}
+                            defaultValue={MODULE_REFERENCE_PLACEHOLDER}
+                            disabled={deriveFromModuleRefernce === DERIVE_FROM_STEP_1.value}
+                            {...form.register('moduleReferenceString', {
+                                required: true,
+                                validate: validateModuleReference,
```

Comment:
Is this validation done on submission, or on every edit?

If it is the former we can make sure its 64 characters.

Reply:
It is done on every edit. The validation function checks both `lenght is 64 characters` and string is a hexstring.
https://github.com/Concordium/concordium-smart-contract-tools/pull/154#discussion_r1492449853